dockers/docker-lldp-sv2/lldpmgrd

```diff
@@ -101,20 +89,41 @@ class LldpManager(object):
     REDIS_TIMEOUT_MS = 0
 
     def __init__(self):
-        # Open a handle to the State database
-        self.state_db = swsscommon.DBConnector(swsscommon.STATE_DB,
-                                               self.REDIS_HOSTNAME,
-                                               self.REDIS_PORT,
-                                               self.REDIS_TIMEOUT_MS)
-
         # Open a handle to the Config database
         self.config_db = swsscommon.DBConnector(swsscommon.CONFIG_DB,
                                                 self.REDIS_HOSTNAME,
                                                 self.REDIS_PORT,
                                                 self.REDIS_TIMEOUT_MS)
 
+        # Open a handle to the Application database
+        self.appl_db = swsscommon.DBConnector(swsscommon.APPL_DB,
+                                                self.REDIS_HOSTNAME,
+                                                self.REDIS_PORT,
+                                                self.REDIS_TIMEOUT_MS)
+
         self.pending_cmds = {}
 
+    def is_port_up(self, port_name):
+        """
+        Determine if a port is up or down by looking into the oper-status for the port in 
+        PORT TABLE in the Application DB
+        """
+        # Retrieve all entires for this port from the Port table
+        port_table = swsscommon.Table(self.appl_db, swsscommon.APP_PORT_TABLE_NAME)
+        (status, fvp) = port_table.get(port_name)
+        if status:
+            # Convert list of tuples to a dictionary
+            port_table_dict = dict(fvp)
+
+            # Get the oper-status for the port
+            port_oper_status = port_table_dict.get("oper_status")
```

Comment:
What if we don't have that value in the dictionary yet?
Please add a default value as a second argument of "oper_status".

Reply:
Good point. If unspecified, the default value is `None`. I'm fine with simply first checking that it si not `None`.

Reply:
@jleveque Yes. You're correct. But anyway I feel it's incorrect semantically to compare None with 'up'.

Reply:
I'm not saying that. I'm saying first check whether the `get()` returns `None`. If not, _then_ compare the value.

Reply:
@jleveque I like your approach better. Thanks